<hr style="height:2px;border:none"/>
<h1 align='center'> INF-280 Estadística Computacional II-2022 </h1>

<H3 align='center'> Tema 7 - Estimación de Intervalos de Confianza  </H3>

\


Nombre del equipo: PIKIBLAINDERS


Integrantes:


*   Ignacio Alvarez - 202073582-4 - [video](https://youtu.be/8RC1QXi5EzM)
*   Nelson Sepúlveda- 202004610-7 - [video](https://youtu.be/tFyiO_fo1WI)
*   Carolina Muñoz - 202004647-6 - [video](https://youtu.be/qigV4tkZeM4)


*Recuerden que si su equipo se llama "El ritmo no perdona", entonces el nombre del archivo a subir debe ser T7_El_ritmo_no_perdona.ipynb*


### **Formalidades**  
* Es posible utilizar apuntes, libros, papers, ejemplos y cualquier otro material que desee de internet, pero debe ser usted quien desarrolle los ejercicios y comente los resultados en el Notebook que entregará. 
* Se permite el uso de las librerías: pandas, numpy, matplotlib y seaborn. Si desea utilizar otra librería debe indicar cómo instalarla.
* Se debe trabajar en grupos de **dos o tres personas**, las cuales deberán estar inscritas a través de __[Aula](https://aula.usm.cl/)__.
* Además del Notebook, deberán entregar un video explicando lo que hizo, paso a paso, como en los ejemplos entregados en AULA. Este vídeo es **individual**, es decir, cada integrante debe hacerlo por separado (debiendo utilizar el mismo Jupyter Notebook). El video puede ser subido a una plataforma externa como Youtube, Google Drive, ó Dropbox. El **link a compartir** (*visible para cualquier usuario con el link*) debe ir dentro del Notebook.
* La nota de esta actividad es grupal.


<hr style="height:2px;border:none"/>

### **Entrega** 
* **Formato**: Se debe realizar una **única** entrega por grupo (un representate del grupo) a través de  __[**Aula**](https://aula.usm.cl/)__.
* **Archivo**: El nombre del archivo a subir debe ser **T7_Nombre_Del_Equipo.ipynb**, por ejemplo, si su equipo se llama "El ritmo no perdona", entonces el nombre del archivo debe ser **T7_El_ritmo_no_perdona.ipynb**
* **Archivo**: Sus respuestas deben ser entregadas en forma de Jupyter Notebook, incluyendo **todos links** a los vídeos explicativos, uno por cada integrante.
    * **Vídeo**: Debe grabar la pantalla (Jupyter Notebok) donde se realiza la explicación de la actividad, con una duración máxima de 25 minutos.
* **Fecha límite**: Domingo 27 de Noviembre a las 23:59

<hr style="height:2px;border:none"/>

### **Descuentos**

* Se descontarán 5 puntos si el nombre del archivo subido no sigue el formato señalado.
* Se descontarán 10 puntos por cada hora de atraso en la entrega.
* Todos los gráficos que realicen deben contener los nombres de las variables en los ejes. Se descontarán 3 puntos por cada nombre faltante.

<hr style="height:2px;border:none"/>

<table align='left'>
<tr>
<td><img src='https://as01.epimg.net/deporteyvida/imagenes/2017/10/28/portada/1509177885_209365_1509178036_noticia_normal.jpg' width='650'/> </td>
</tr>
</table>

## Contexto

Las enfermedades cardiovasculares (ECV) son la principal causa de muerte en todo el mundo y se cobran aproximadamente 17,9 millones de vidas al año, lo que representa el 31 % de todas las muertes en todo el mundo. Cuatro de cada 5 muertes por ECV se deben a ataques cardíacos y accidentes cerebrovasculares, y un tercio de estas muertes ocurren prematuramente en personas menores de 70 años. 

La insuficiencia cardíaca es un evento común causado por ECV. Por esto, se ha construido un dataset llamado `heart.csv`, el cual contiene 11 características que pueden usarse para predecir una posible enfermedad cardíaca. Dichas características se detallan a continuación:


1.   `Age`: age of the patient [years]
2.   `Sex`: sex of the patient [M: Male, F: Female]
3.   `ChestPainType`: chest pain type [TA: Typical Angina, ATA: Atypical Angina, NAP: Non-Anginal Pain, ASY: Asymptomatic]
4.   `RestingBP`: resting blood pressure [mm Hg]
5.   `Cholesterol`: serum cholesterol [mm/dl]
6.   `FastingBS`: fasting blood sugar [1: if FastingBS > 120 mg/dl, 0: otherwise]
7.   `RestingECG`: resting electrocardiogram results [Normal: Normal, ST: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV), LVH: showing probable or definite left ventricular hypertrophy by Estes' criteria]
8.   `MaxHR`: maximum heart rate achieved [Numeric value between 60 and 202]
9.   `ExerciseAngina`: exercise-induced angina [Y: Yes, N: No]
10.  `Oldpeak`: oldpeak = ST [Numeric value measured in depression]
11.  `ST_Slope`: the slope of the peak exercise ST segment [Up: upsloping, Flat: flat, Down: downsloping]
12.  `HeartDisease`: output class [1: heart disease, 0: Normal]




# Librerías

In [6]:
import pandas as pd
import numpy as np
import seaborn as sns; sns.set()
import matplotlib.pyplot as plt
import warnings
import scipy.stats as st
warnings.filterwarnings("ignore")


# Datos

In [7]:
data = pd.read_csv("heart.csv")
data

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  \
0     40   M           ATA        140          289          0     Normal   
1     49   F           NAP        160          180          0     Normal   
2     37   M           ATA        130          283          0         ST   
3     48   F           ASY        138          214          0     Normal   
4     54   M           NAP        150          195          0     Normal   
..   ...  ..           ...        ...          ...        ...        ...   
913   45   M            TA        110          264          0     Normal   
914   68   M           ASY        144          193          1     Normal   
915   57   M           ASY        130          131          0     Normal   
916   57   F           ATA        130          236          0        LVH   
917   38   M           NAP        138          175          0     Normal   

     MaxHR ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0      172              N      0.0       Up             0  
1      156              N      1.0     Flat             1  
2       98              N      0.0       Up             0  
3      108              Y      1.5     Flat             1  
4      122              N      0.0       Up             0  
..     ...            ...      ...      ...           ...  
913    132              N      1.2     Flat             1  
914    141              N      3.4     Flat             1  
915    115              Y      1.2     Flat             1  
916    174              N      0.0     Flat             1  
917    173              N      0.0       Up             0  

[918 rows x 12 columns]

# Preguntas

Para las siguientes preguntas considere un nivel de confianza del **$95\%$**, a menos que se indique lo contrario.


1. Se piensa que la edad es un factor importante a la hora de determinar si una persona tiene una enfermedad al corazón(HeartDisease=1). En particular, se cree que las mujeres tienden a presentar una enfermedad al corazón a una edad mayor que los hombres. ¿Se puede afirmar que la edad media de las mujeres que tienen una enfermedad al corazón es mayor que la edad media de los hombres con una enfermedad al corazón? **(15 pts)**
2. Determine qué ocurre con su respuesta de la pregunta anterior a medida que $α$ aumenta entre 0 y 1. **(10 pts)**
3. Se está realizando una campaña preventiva contra las enfermedades al corazón, donde se ha observado que estas enfermedades afectan principalmente a personas mayores de 50 años. Sin embargo, se busca determinar si afecta más a los hombres o a las mujeres, o bien afecta a ambos sexos por igual. Considerando solamente a las personas mayores de 50 años, determine si la proporción de hombres con enfermedades al corazón es mayor que la proporción de mujeres con enfermedades al corazón. **(15 pts)**
4. Determine si la media del colesterol de los hombres "sanos"(HeartDisease=0) es mayor que la media del colesterol de las mujeres "sanas". Además, analice lo que ocurre con sus varianzas y concluya al respecto. **(20 pts)**. **IMPORTANTE: No considere los datos que tengan Cholesterol=0**
5. Se cree que la media de la presión arterial durante el reposo (RestingBP) suele ser mayor en personas con enfermedades al corazón (HeartDisease=1). Para las siguiente preguntas **considere un nivel de confianza del $99.5\%$. Además, No considere los datos que tengan RestingBP=0**

    > a) Determine si la afirmación es verdadera considerando solamente a las mujeres. En caso de no ser cierta, ¿Cuál es el valor mínimo de $\alpha$ para que la afirmación sea verdadera? **(5 pts)**

    > b) Determine si la afirmación es verdadera considerando solamente a los hombres. En caso de no ser cierta, ¿Cuál es el valor mínimo de $\alpha$ para que la afirmación sea verdadera? **(5 pts)**

    > c) Determine si la afirmación es verdadera para todas las personas. En caso de no ser cierta, ¿Cuál es el valor mínimo de $\alpha$ para que la afirmación sea verdadera? **(5 pts)**

6.  El dolor en el pecho al realizar ejercicios (ExerciseAngina='Y') es un síntoma que se suele presentar en personas con enfermedades al corazón, ya que el corazón de estas personas no recibe suficiente sangre oxigenada.
    
    > a) Considerando solamente a las personas **con** enfermedades al corazón, ¿Se puede afirmar que la proporción de personas con dolor en el pecho cuando realizan ejercicios (ExerciseAngina='Y')` es mayor al 50%?, ¿Cuál es el máximo nivel de confianza con el cual se cumple esta afirmación? **(15 pts)**

    > b) ¿Qué ocurre si solamente se consideran a las mujeres con enfermedades al corazón en la pregunta a)? **(5 pts)**

    > c) ¿Qué ocurre si solamente se consideran a los hombres con enfermedades al corazón en la pregunta a)? **(5 pts)**



### Pregunta 1

Para responder esta pregunta calcularemos la media muestral y un intervalo de confianza para el valor de la media poblacional. A partir de ambos datos concluiremos la veracidad de la afirmación planteada en la pregunta.

In [8]:
#DataFrame de Hombres
data_h = data[data.Sex == 'M']

#DataFrame Hombres con HeartDisease = 1.
data_hHD = data_h[data_h.HeartDisease == 1]
print("Hombres: ", st.norm.interval(alpha = 0.95, loc = np.mean(data_hHD['Age']), scale = st.sem(data_hHD['Age'])))


#DataFrame de Mujeres
data_m = data[data.Sex == 'F']

#DataFrame Mujeres con HeartDisease = 1.
data_mHD = data_m[data_m.HeartDisease == 1]
print("Mujeres: ", st.norm.interval(alpha = 0.95, loc = np.mean(data_mHD['Age']), scale = st.sem(data_mHD['Age'])))

print("\nMedia poblacional para hombres: ",np.mean(data_hHD['Age']))
print(  "Media poblacional para mujeres: ",np.mean(data_mHD['Age']))

Hombres:  (55.06411188020128, 56.673879386174264)
Mujeres:  (53.90139230573687, 58.45860769426313)

Media poblacional para hombres:  55.86899563318777
Media poblacional para mujeres:  56.18


Analizando los intervalos podemos ver que estos se intersectan, por lo que es poco preciso hacer alguna afirmacion acerca del valor de la media. Sin embargo sabemos que (gracias al metodo de los momentos)la media muestral es un buen estimador de la media poblacional, por lo que se podría decir que la afirmacion de que la edad media de las mujeres que tienen una enfermedad al corazón es mayor que la edad media de los hombres con una enfermedad al corazón.

## Pregunta 2

"$α$" es conocido como la significancia de un I.C, y éste define el nivel de confianza del I.C, de forma que si aumenta el valor de $α$ disminuye el valor del nivel de confianza y vice versa.

In [9]:
print("Hombres con alpha=0.005: ", st.norm.interval(alpha = 0.995, loc = np.mean(data_hHD['Age']), scale = st.sem(data_hHD['Age'])))
print("Mujeres con alpha=0.005: ", st.norm.interval(alpha = 0.995, loc = np.mean(data_mHD['Age']), scale = st.sem(data_mHD['Age'])))

print("\nHombres con alpha=0.10: ", st.norm.interval(alpha = 0.90, loc = np.mean(data_hHD['Age']), scale = st.sem(data_hHD['Age'])))
print(  "Mujeres con alpha=0.10: ", st.norm.interval(alpha = 0.90, loc = np.mean(data_mHD['Age']), scale = st.sem(data_mHD['Age'])))

print("\nHombres con alpha=0.50: ", st.norm.interval(alpha = 0.50, loc = np.mean(data_hHD['Age']), scale = st.sem(data_hHD['Age'])))
print(  "Mujeres con alpha=0.50: ", st.norm.interval(alpha = 0.50, loc = np.mean(data_mHD['Age']), scale = st.sem(data_mHD['Age'])))

print("\nHombres con alpha=0.75: ", st.norm.interval(alpha = 0.25, loc = np.mean(data_hHD['Age']), scale = st.sem(data_hHD['Age'])))
print(  "Mujeres con alpha=0.75: ", st.norm.interval(alpha = 0.25, loc = np.mean(data_mHD['Age']), scale = st.sem(data_mHD['Age'])))

print("\nHombres con alpha=1: ", st.norm.interval(alpha = 0, loc = np.mean(data_hHD['Age']), scale = st.sem(data_hHD['Age'])))
print(  "Mujeres con alpha=1: ", st.norm.interval(alpha = 0, loc = np.mean(data_mHD['Age']), scale = st.sem(data_mHD['Age'])))

Hombres con alpha=0.005:  (54.71625206644587, 57.021739199929684)
Mujeres con alpha=0.005:  (52.916609043300554, 59.44339095669945)

Hombres con alpha=0.10:  (55.19351589436756, 56.54447537200799)
Mujeres con alpha=0.10:  (54.26773214208434, 58.09226785791566)

Hombres con alpha=0.50:  (55.59200797128635, 56.145983295089195)
Mujeres con alpha=0.50:  (55.39585419598459, 56.96414580401541)

Hombres con alpha=0.75:  (55.73814238836294, 55.99984887801261)
Mujeres con alpha=0.75:  (55.8095574244469, 56.5504425755531)

Hombres con alpha=1:  (55.86899563318777, 55.86899563318777)
Mujeres con alpha=1:  (56.18, 56.18)


Podemos observar que cuando $α$ toma valores cercanos a 0 los intervalos son más anchos, y a medida que $α$ toma valores cercanos a 1 los intervalos son más estrechos. Esto es coherente con lo mencionado anteriormente, dado que el hecho de disminuir el valor de $α$ implica un aumento en el nivel de confianza, siendo el nivel de confianza un indicador de que tan posible es encontrar un parametro dentro del intervalo, por lo que es logico que al aumentar el largo del intervalo también aumente en nivel de confianza.

# Pregunta 3

Calcularemos un intervalo de confianza con un nivel de confianza del 95% para la media de enfermedades al corazon de la poblacion mayor de 50 años, separando entre hombres y mujeres.
Destacar que utilizamos la media como indicador del porcentaje de hombres/mujeres con enfermedades al corazon dado que el dominio de esta variable es solo 0 y 1, lo que produce que la media sea equivalente a la proporcion de la poblacion que tenga algun padecimiento.

In [10]:
# Intervalo para los hombres
data_men = data[data.Sex =='M']
data_men = data_men[data_men.Age >50]

st.norm.interval(alpha = 0.95, loc = np.mean(data_men['HeartDisease']), scale = st.sem(data_men['HeartDisease']))

(0.6658754603404021, 0.7468555458197622)

In [11]:
# Intervalo para las mujeres
data_fem = data[data.Sex =='F']
data_fem = data_fem[data_fem.Age > 50]

st.norm.interval(alpha = 0.95, loc = np.mean(data_fem['HeartDisease']), scale = st.sem(data_fem['HeartDisease']))

(0.2522270034512278, 0.4260338661139896)

Con estos resultados es clara la conclusión de que la proporción de hombres con enfermedades al corazón es mayor que la proporción de mujeres con enfermedades al corazón.

# Pregunta 4

En esta pregunta, al igual que en la 1, realizaremos el calculo de las medias poblacionales y un intervalo de confianza con un nivel de confianza del 95% para concluir si la media del colesterol de los hombres "sanos" es mayor que la media del colesterol de las mujeres "sanas".
Tambien calcularemos 

In [12]:
# Intervalo para los hombres
data_men = data[data.Sex =='M']
data_men = data_men[data_men.HeartDisease == 0]
data_men = data_men[data_men.Cholesterol != 0]

print("Media muestrales para el colesterol en hombres: ",np.mean(data_men['Cholesterol']))

print("\nIntervalo de confianza para la media de colesterol en hombres: ",st.norm.interval(alpha = 0.95, loc = np.mean(data_men['Cholesterol']), scale = st.sem(data_men['Cholesterol'])))

Media muestrales para el colesterol en hombres:  232.80645161290323

Intervalo de confianza para la media de colesterol en hombres:  (226.55364732109453, 239.05925590471193)


In [13]:
# Intervalo para las mujeres
data_fem = data[data.Sex =='F']
data_fem = data_fem[data_fem.HeartDisease == 0]
data_fem = data_fem[data_fem.Cholesterol != 0]

print("Media muestral para el colesterol en mujeres: ",np.mean(data_fem['Cholesterol']))

print("\nIntervalo de confianza para la media de colesterol en mujeres: ",st.norm.interval(alpha = 0.95, loc = np.mean(data_fem['Cholesterol']), scale = st.sem(data_fem['Cholesterol'])))

Media muestral para el colesterol en mujeres:  249.18309859154928

Intervalo de confianza para la media de colesterol en mujeres:  (238.94800137283738, 259.4181958102612)


In [14]:
# I.C para la varianza en hombres
n_men = len(data_men.index)-1 # grados de libertad

s = np.std(data_men['Cholesterol'])**2 # varianza de la muestra

Limite_inf = st.chi2.ppf(0.975, n_men)
Limite_sup = st.chi2.ppf(0.025, n_men)

Limite_sup = round((n_men*s)/Limite_sup,2)
Limite_inf = round((n_men*s)/Limite_inf,2)
print("Intervalo de confianza para la varianza en hombres: (",Limite_inf,",", Limite_sup, ")")

Intervalo de confianza para la varianza en hombres: ( 2123.41 , 3023.62 )


In [15]:
# I.C para la varianza en mujeres
n_fem = len(data_fem.index)-1 # grados de libertad

s = np.std(data_fem['Cholesterol'])**2 # varianza de la muestra

Limite_inf = st.chi2.ppf(0.975, n_fem)
Limite_sup = st.chi2.ppf(0.025, n_fem)

Limite_sup = round((n_fem*s)/Limite_sup,2)
Limite_inf = round((n_fem*s)/Limite_inf,2)
print("Intervalo de confianza para la varianza en mujeres: (",Limite_inf, ",", Limite_sup, ")")

Intervalo de confianza para la varianza en mujeres: ( 3084.56 , 4927.82 )


Para el caso de las medias podemos ver que los intervalos se intersectan, por lo que es poco preciso concluir a partir de los estos. Sin embargo para valor puntual estimado podemos ver que la media de colesterol en mujeres es notablemente más alta a pesar de que en los intervalos haya interseccion. Esto último puede ser explicado a traves del valor de la varianza de ambas poblaciones, siendo la de las mujeres indiscutiblemente mayor a la de los hombres, cosa que implica que los intervalos de confianza para la media sean más anchos en mujeres que en hombres a pesar de tener el mismo nivel de confianza, causando así que a pesar de tener valores estimados alejados, los intervalos de confianza para la media se intersecten.

## Pregunta 5

### 5.a) Considerando solo a Mujeres.

In [83]:
#DataFrame filtrando solo cuando son Mujeres.
data_fem = data[data.Sex == 'F']

#DataFrame filtrando cuando son Mujeres con HeartDisease = 1.
data_fem_1 = data_fem[data_fem.HeartDisease == 1]
#DataFrame sin contar los RestingBP == 0.
data_fem_1R = data_fem_1[data_fem.RestingBP != 0]
st.norm.interval(alpha = 0.995, loc = np.mean(data_fem_1R['RestingBP']), scale = st.sem(data_fem_1R['RestingBP']))

#print(data_fem_1)

(132.77270559747782, 151.2272944025222)

In [84]:
#DataFrame filtrando solo cuando son Mujeres.
data_fem = data[data.Sex == 'F']

#DataFrame filtrando cuando son Mujeres con HeartDisease = 0.
data_fem_0 = data_fem[data_fem.HeartDisease == 0]
#DataFrame sin contar los RestingBP == 0.
data_fem_0R = data_fem_0[data_fem_0.RestingBP != 0]
st.norm.interval(alpha = 0.995, loc = np.mean(data_fem_0R['RestingBP']), scale = st.sem(data_fem_0R['RestingBP']))

#print(data_fem_0)

(124.88671851267976, 132.69370106773982)

Luego, el intervalo de confianza que resulto:

Para Mujeres CON enfermedades al corazón(HeartDisease=1) = (132.34, 151.66).  
Para Mujeres SIN enfermedades al corazon(HeartDisease=0) = (124.83, 132.76).

Se observa claramente que el intervalo de confiaza de la media de la presión arterial durante el reposo suele ser mayor en mujeres CON enfermedades al corazón, es decir esta afirmación es verdadera para este caso, con un alpha minimo de 0,995.

### 5.b) Considerando solo a Hombres.

In [96]:
#DataFrame filtrando solo cuando son Hombres.
data_mas = data[data.Sex == 'M']

#DataFrame filtrando cuando son Hombres con HeartDisease = 1.
data_mas_1 = data_mas[data_mas.HeartDisease == 1]
#DataFrame sin contar los RestingBP == 0.
data_mas_1R = data_mas_1[data_mas_1.RestingBP != 0]
st.norm.interval(alpha = 0.8508, loc = np.mean(data_mas_1R['RestingBP']), scale = st.sem(data_mas_1R['RestingBP']))

#print(data_mas_1)

(132.3931532670963, 134.8541115031444)

In [97]:
#DataFrame filtrando solo cuando son Hombres.
data_mas = data[data.Sex == 'M']

#DataFrame filtrando cuando son Hombres con HeartDisease = 0.
data_mas_0 = data_mas[data_mas.HeartDisease == 0]
#DataFrame sin contar los RestingBP == 0.
data_mas_0R = data_mas_0[data_mas_0.RestingBP != 0]
st.norm.interval(alpha = 0.8508, loc = np.mean(data_mas_0R['RestingBP']), scale = st.sem(data_mas_0R['RestingBP']))

#print(data_mas_0)

(129.47634246356455, 132.37384480235306)

Luego, el intervalo de confianza que resulto:

Para Hombres CON enfermedades al corazón(HeartDisease=1) = (132.178, 134.485).  
Para Hombres SIN enfermedades al corazon(HeartDisease=0) = (129.636, 132.214).

Se observa claramente que el intervalo de confiaza de la media de la presión arterial durante el reposo 
suele ser mayor en HOmbres CON enfermedades al corazón, es decir esta afirmación es verdadera para un alpha minimo de 0,8508(85,08%).


### 5.c) Considerando a todas las personas.

In [98]:
#DataFrame filtrando cuando HeartDisease = 1.
data_all_1 = data[data.HeartDisease == 1]
#DataFrame sin contar los RestingBP == 0.
data_all_1R = data_all_1[data_all_1.RestingBP != 0]

#Intervalos de confiaza respecto del RestingBP para Todas las personas, es decir, usando todos los datos de la tabla original
st.norm.interval(alpha = 0.97, loc = np.mean(data_all_1R['RestingBP']), scale = st.sem(data_all_1R['RestingBP']))



(132.62540166684235, 136.2740066171813)

In [99]:
#DataFrame filtrando cuando HeartDisease = 0.
data_all_0 = data[data.HeartDisease == 0]
#DataFrame sin contar los RestingBP == 0.
data_all_0R = data_all_0[data_all_0.RestingBP != 0]

#Intervalos de confiaza respecto del RestingBP para Todas las personas, es decir, usando todos los datos de la tabla original
st.norm.interval(alpha = 0.97, loc = np.mean(data_all_0['RestingBP']), scale = st.sem(data_all_0['RestingBP']))


(128.4121757637281, 131.948799846028)

Finalmente, el intervalo de confianza que resulto:

Para Personas CON enfermedades al corazón(HeartDisease=1) = (132.27, 136.09).

Para Personas SIN enfermedades al corazon(HeartDisease=0) = (128.41, 131.96).

Se observa claramente que el intervalo de confiaza de la media de la presión arterial durante el reposo 
suele ser mayor en personas CON enfermedades al corazón, es decir esta afirmación es verdadera para un alpha minimo de 0,97(97%).

### 6.a) Considerando a todas las personas.

La proporciones en estadistica se calculan de la siguiente manera: p = x/N.

x: la cantidad de exitos.

N: el numero total de pruebas.

In [44]:
#DataFrame considerando solo HeartDisease = 1.
data_allsick = data[data.HeartDisease == 1]

#luego, considerando como el total todas las personas con HeartDisease = 1 y como casos de exito ExerciseAngina = Y
N = len(data_allsick['ExerciseAngina']) #508
x = len(data_allsick[data_allsick.ExerciseAngina == 'Y']['ExerciseAngina']) #316
y = len(data_allsick[data_allsick.ExerciseAngina == 'N']['ExerciseAngina']) #192


P = (x/N)

info = []
for i in range(y):
    info.append(0)

for i in range(x):
    info.append(1)

st.norm.interval(alpha = 0.97, loc = P, scale = st.sem(info))

Porcentaje % 62.2


El porcentaje de la proporción de personas con enfermedades al corazón, con dolor en el pecho cuando realizan ejercicios es 62,2% el cual es mayor al 50%, por lo que esta afirmacion es correcta.

### 6.b) Considerando solo a Mujeres

In [45]:
#DataFrame considerando solo HeartDisease = 1.
data_allsick = data[data.HeartDisease == 1]

#considerando solo a las mujeres
data_womensick = data_allsick[data_allsick.Sex == 'F']

#luego, considerando como el total todas las personas con HeartDisease = 1 y como casos de exito ExerciseAngina = Y
N = len(data_womensick['ExerciseAngina']) #50
x = len(data_womensick[data_womensick.ExerciseAngina == 'Y']['ExerciseAngina']) #27

P = (x/N)*100
print("Porcentaje de proporción: %",round(P,2))

Porcentaje de proporción: % 54.0


El porcentaje de la proporción de Mujeres con enfermedades al corazón, con dolor de pecho cuando realizan ejercicio es de un 54%, por lo que, esta afirmación también es verdadera.

### 6.C)

In [100]:
#DataFrame considerando solo HeartDisease = 1.
data_allsick = data[data.HeartDisease == 1]

#considerando solo a los hombres
data_mensick = data_allsick[data_allsick.Sex == 'M']

#luego, considerando como el total todas las personas con HeartDisease = 1 y como casos de exito ExerciseAngina = Y
N = len(data_mensick['ExerciseAngina']) #458
x = len(data_mensick[data_mensick.ExerciseAngina == 'Y']['ExerciseAngina']) #289

P = (x/N)*100
print("Porcentaje de proporción: %",round(P,2))

Porcentaje de proporción: % 63.1


la proporcion de Hombres enfermos del corazón, con dolor de pecho cuando realizan ejercicio es de un 63,1% por lo que, al igual que el resto de las afirmaciones es verdadera.